lib/app/data/post_client.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:black_tax_and_white_benefits/app/domain/post.dart';
+import 'package:dio/dio.dart';
+import 'package:retrofit/http.dart';
+import 'package:retrofit/retrofit.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'post_client.g.dart';
+
+@RestApi(baseUrl: 'https://blacktaxandwhitebenefits.com/wp-json/wp/v2/')
+abstract class PostClient {
+  factory PostClient(Dio dio, {String baseUrl}) = _PostClient;
+
+  @GET('/posts?_embed=true')
+  Future<List<Post>> getPosts(
+    @Query('per_page') int n,
+  );
+}
+
+// coverage:ignore-start
+@riverpod
+FutureOr<List<Post>> getPosts(GetPostsRef ref) async {
```

Comment:
@alexdivadi  plz make a jira task to get coverage for this, perhaps with fakedio.
the imporatnt thing to keep in mind about fakedio is that his calls use dio.get etc and retrofit uses fetch. (see the .g file). other than that it should be all good, and if you get stuck i can finish from your branch.

Reply:
Link here: [ticket](https://anirac-tech.atlassian.net/browse/WA-22?atlOrigin=eyJpIjoiY2QyMDBkZGM2OWU1NDRlOTg2ZDY0ZGI1NDFiMjkzMGUiLCJwIjoiaiJ9)